### ISSUE TITLE
 user-keys bindings not working

### ISSUE THREAD

**Issue by Federico-Ciuffardi** [tmux/tmux#2658]
### Issue description
All my user-keys bindings stopped working on the latest update,  just cloned the repo and installed tmux from source, But the user-keys bindings still not working. 

it works for C-Tab and C-S-t, but:
    * C-S-J binding triggers when C-S-K is pressed and vice versa
    * C-S-Enter does not work
    * Can't find how to bind to some keys combinations, like for example: C--, C-`, C-S-Backspace

### Required information

Please provide the following information:

* tmux version: tmux next-3.3
* Platform: Linux unknown (I use arch btw)
* $TERM inside tmux: st-256color
* $TERM outside tmux: st-256color
* Logs from tmux: https://github.com/Federico-Ciuffardi/tmux-logs

The logs correspond to the execution of `tmux kill-server; tmux -vv new` and then the following key presses: C-Tab, C-S-t, C-S-J, C-S-K,  C-S-Enter, C--, C-` and C-S-Backspace.

**Comment by nicm** [comment]
These seem OK:
~~~
1618590489.998213 /dev/pts/3: keys are 6 (\033[9;5u)       <-- C-Tab
1618590491.439037 /dev/pts/3: keys are 7 (\033[84;6u)      <-- C-S-t
1618590492.357031 /dev/pts/3: keys are 7 (\033[75;6u)      <-- C-S-k
1618590492.816132 /dev/pts/3: keys are 7 (\033[74;6u)      <-- C-S-j
1618590496.190098 /dev/pts/3: keys are 7 (\033[45;5u)      <-- C--
1618590497.672030 /dev/pts/3: keys are 7 (\033[96;5u)      <-- C-`
1618590499.367952 /dev/pts/3: keys are 8 (\033[127;6u)     <-- C-S-BSpace
~~~~
What is the problem with binding these?

This one tmux sees as `S-Enter` which is a bug:
~~~~
1618590494.130293 /dev/pts/3: keys are 7 (\033[13;6u)
~~~~

**Comment by Federico-Ciuffardi** [comment]
* C-S-J binding triggers when C-S-K is pressed and vice versa
1618590492.357031 /dev/pts/3: keys are 7 (\033[75;6u)      <-- C-S-k  (here I pressed C-S-j)
1618590492.816132 /dev/pts/3: keys are 7 (\033[74;6u)      <-- C-S-j   (here I pressed C-S-k)

* Can't find how to bind to some keys combinations, like for example: C-`
When the user-keys bindings were working this worked fine, but now I don't know how to bind these keys. Binding these results in unknown key when reloading the config.

Will the user-keys bindings work in the future? I ask this because I would like to avoid modifying my config if the fix for user-keys is on the way?

**Comment by nicm** [comment]
tmux has j and k correctly by the look of it, if they are the wrong way round it seems like a bug in your terminal.

The keys bind OK for me but I can't guess what you are putting in .tmux.conf.

**Comment by nicm** [comment]
I might make it so `user-keys` lookup is before extended keys but not sure when.

**Comment by Federico-Ciuffardi** [comment]
> I might make it so user-keys lookup is before extended keys but not sure when.

* Ok! I will try to update my config then.

> tmux has j and k correctly by the look of it, if they are the wrong way round it seems like a bug in your terminal.

* I fixed that on my terminal! 

* Is there a document where I can see the keycodes - key press correspondence that tmux is using? 
(so I can fix other similar bugs that might arise)

> The keys bind OK for me but I can't guess what you are putting in .tmux.conf.

* In my .tmux.conf I have the following lines:
```
bind-key -n C-` run "$TMUX_BIN/switch_session_detached '#{session_id}'" 
bind-key -n C-/ split-window -h -c "#{pane_current_path}"
```
And loading the config results on
```
Unknown key:  C-`
Unknown key:  C-/
```

Let me know if you need the full config!

* Also is there a documentation on how to the keys are named on tmux? 
I did not know that the backspace was named BSpace, also **I do not know how to refer to Escape**.

### Unrelated comment
C-S-M gets triggered by pressing C-S-M and C-S-Enter

**Comment by nicm** [comment]
Put C-/ and C-` in single quotes.

The man page has the special keys. Escape is `Escape`. But I don't think we support `C-Escape`, I will need to fix that.

For these extended key sequences. 75 = ASCII j, 76 = ASCII k. See http://www.leonerd.org.uk/hacks/fixterms/.

**Comment by nicm** [comment]
I will take a look at C-S-Enter and modifiers with Escape next week.

**Comment by Federico-Ciuffardi** [comment]
> Put C-/ and C-` in single quotes.

Did not work, chaged the lines to:
```
bind-key -n C-` run "$TMUX_BIN/switch_session_detached '#{session_id}'" 
bind-key -n 'C-/' split-window -h -c "#{pane_current_path}"
```
But stills throws the error Unknown key in both cases.

> The man page has the special keys. Escape is Escape. But I don't think we support C-Escape, I will need to fix that.
> For these extended key sequences. 75 = ASCII j, 76 = ASCII k. See http://www.leonerd.org.uk/hacks/fixterms/.
> I will take a look at C-S-Enter and modifiers with Escape next week.

Thank you!

**Comment by nicm** [comment]
Oh yeah sorry, we don't have those keys. I will look at that too.

**Comment by Federico-Ciuffardi** [comment]
Ok, I will wait!

Thank you very much for your quick responses and for creating tmux!

**Comment by nicm** [comment]
Can you try this please and see if any of your problems remain: [x.diff.txt](https://github.com/tmux/tmux/files/6331454/x.diff.txt)

**Comment by Federico-Ciuffardi** [comment]
Nice! Everything but the following line is working:
```
bind-key -n 'C-/' split-window -h -c "#{pane_current_path}"
```
When pressing `C-/` Instead of the expected action, the following gets outputted on the terminal `^_`

And these are the keys logged on the tmux-server log (output of `grep "keys are" tmux-server-515387.log`):
```
1618856417.285869 /dev/pts/7: keys are 1 (\037)
1618856418.995922 /dev/pts/7: keys are 1 (\037)
1618856420.335231 /dev/pts/7: keys are 3 (\033[O)
1618856421.531942 /dev/pts/3: keys are 1 (p)
1618856421.719925 /dev/pts/3: keys are 1 (k)
1618856421.890994 /dev/pts/3: keys are 1 (i)
1618856422.244937 /dev/pts/3: keys are 1 (\005)
1618856422.514903 /dev/pts/3: keys are 1 (\r)
```

Tell me if you need any more information.

**Comment by nicm** [comment]
That is correct, `C-/` and `C-_` are the same. You will need to bind `C-_` since your terminal does not send anything different for `C-/`.

**Comment by Federico-Ciuffardi** [comment]
Everything is working now.

Thank you very much!

**Comment by nicm** [comment]
Great, applied, thanks!
